### DISCUSSION TITLE
Standard vs. Open Formats

### DISCUSSION THREAD

**Discussion by ALRubinger** [TBD54566975/open-source-programs#7] [General]
In our discussion about [decomposing the system and decentralization](https://github.com/TBD54566975/collaboration/discussions/5#discussioncomment-2249289), @mistermoe proposed 5 aspects of decentralized systems, and encouraged us to view "decentralization" as a spectrum.

I'd like us to dig into one of them here: "standards-based formats". @mistermoe cites:

> The benefit of all data adhering to a format specified in a standard is ease of portability

@csuwildcat also takes the stance:

> Whatever we develop must be based on standard specifications that are in (at very least) a draft state (or equivalent) within an appropriate standard organization. If our work is not based on open standards, it is just another corporate project that happens to have a permissive license.

I offer a tweak:

* Standards *strengthen* the portability guarantee
* Standards are expensive. That strength has a cost, and the cost is time and coordination with a standards body.
* Standards done prematurely, before proven in the field, are brittle

*I propose making our message formats and protocols an open format*. As we prove its worth in the field and assemble buy-in from the community, we may pursue parking it in a standards body to strengthen the portability guarantee.

IMO standards bodies are for hardening, not for innovation.

Some examples of open formats that preceded standards:

* The Docker container format
* Hibernate ORM Mapping
* OGG Vorbis and FLAC audio formats

I think this approach will free us to move faster without dropping portability for our users.

**Comment by thereisnogabe** [comment]
Agreed. My thoughts:
- Use existing standards
- Build open formats where there's a gap
- Standardize those formats once they have become undeniable

**Reply by ALRubinger** [reply]
I think this is a concise distillation guiding our approach to standards and open formats. Enough so that if we all agree, I'd like to encode it in a "Guiding Principles" page.

Anyone with counter arguments to make?

**Reply by csuwildcat** [reply]
Not entirely a counter, but a modification: https://github.com/TBD54566975/collaboration/discussions/7#discussioncomment-2266311

**Comment by csuwildcat** [comment]
If there are standards efforts that cover the same things we need, we should be using/advancing those standards. Creating something random/divergent should be the last resort after lengthy review of existing standards options and initiatives.

**Reply by thereisnogabe** [reply]
Agree. Does this capture the revision?

> Use existing standards; where there's a gap, contribute to them
> Build open formats where there aren't suitable standards
> Standardize those formats once they have become undeniable
